xc7/utils/prjxray_form_channels.py

```diff
@@ -634,6 +929,31 @@ def classify_nodes(conn):
     write_cur.connection.commit()
 
 
+def get_node_rc(conn, node_pkey):
+    """ Returns capacitance and resistance for given node.
+
+    Returns
+    -------
+    capacitance : float
+        Node capacitance (Farads)
+    resistance : float
+        Node resistance (Ohms)
+
+    """
+    capacitance = 0
+    resistance = 0
+
+    cur = conn.cursor()
+    for wire_cap, wire_res in cur.execute("""
+SELECT capacitance, resistance FROM wire_in_tile WHERE pkey IN (
+    SELECT wire_in_tile_pkey FROM wire WHERE node_pkey = ?
+    );""", (node_pkey, )):
+        capacitance += wire_cap
```

Comment:
This calculation is wrong from the electrical point of view. The actual R and C of a node will depend on its physical position and length and also which "to" and "from" edges are considered.

Reply:
This is not an electrical model, but a timing model.  This is in fact the correct timing model.  I've created an Excel based timing model for approximating the Vivado model ( https://github.com/SymbiFlow/prjxray/blob/master/prjxray/timing.py ), and it matches Vivado's timing model +/- 1.25 %